sources.go

```diff
@@ -102,3 +103,47 @@ func (s pstateSource) Discover() ([]string, error) {
 
 	return features, nil
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// Network Source
+
+// Implements main.FeatureSource.
+type networkSource struct{}
+
+func (s networkSource) Name() string { return "netid" }
+func (s networkSource) Discover() ([]string, error) {
+	features := []string{}
+	var total_num_vfs int = 0
+	// sysfs of the node mounted as /hostsys inside the pod
+	netInterfaces, err := ioutil.ReadDir("/hostsys/class/net") //does not return . and ..
+	if err != nil {
+		return nil, fmt.Errorf("can't obtain the network interfaces: %s", err.Error())
```

Comment:
Ok. But I would like the error message to be specific to be helpful for debugging. May be something more specific like `can't obtain network interface details from sysfs` might be better?